include/net/net_pkt.h

```diff
@@ -239,7 +239,22 @@ struct net_pkt {
 #endif /* CONFIG_NET_IPV6 */
 
 #if defined(CONFIG_IEEE802154)
-	uint8_t ieee802154_rssi; /* Received Signal Strength Indication */
+	union {
+		uint8_t ieee802154_rssi; /* Received Signal Strength Indication */
+#if defined(CONFIG_IEEE802154_SELECTIVE_TXPOWER)
+		int8_t ieee802154_txpwr; /* TX power. It should be clear from
+					  * the context which field of the union
+					  * is valid at the moment.
+					  */
+	};
+	uint8_t ieee802154_txpwr_use : 1; /* Indicates if provided TX power should
```

Comment:
Position in structure. Could you please check (possibly with godbolt.org) if putting this field here instead of mixed sowewhere with other bitfields just after `ieee802154_mac_hdr_rdy` requires additional allocation unit?
See https://en.cppreference.com/w/c/language/bit_field

Reply:
The structure has multiple variants controlled by Kconfig options, so the answer is not obvious. However, I've checked the part under `#if defined(CONFIG_IEEE802154)` and in some cases rearranging the `ieee802154_txpwr_use` field saves 1B. It's not a staggering saving, but there might be many variables of this type alive at a time, each wasting this one byte.

I'll rearrange the 802.15.4-related fields to shrink the structure's size.

Reply:
Could an invalid `ieee802154_txpwr` value be reserved instead of `ieee802154_txpwr_use` to simplify the code?

Reply:
@edmont I've considered using some specific value of `ieee802154_txpwr` to indicate that it should be ignored, yes. However, in my opinion it doesn't simplify the code really. Here goes the explanation:

Let's assume that the nRF IEEE 802.15.4 radio driver exposes a macro, which can be used in `ieee802154_nrf5.c` to indicate that a given transmission should use the default TX power. That macro is only known to `ieee802154_nrf5.c`. Higher layers such as Openthread radio platform implementation are not aware what exact hardware they're being executed on - they call ieee802154 APIs through pointers. So yet another macro would have to be introduced, so that they can be mapped into each other. All other vendors would have to use the same constant too. The flag is easier to pass up and down between layers.